src/NHibernate.Test/Linq/WhereTests.cs

```diff
@@ -738,6 +738,28 @@ public void BitwiseQuery3()
 			Assert.That(query.Count, Is.EqualTo(3));
 		}
 
+		[Test(Description = "GH-3156")]
+		public void CanUseNullableWithoutAValue()
+		{
+			bool? isActive = null;
+
+			Assert.DoesNotThrow(
+				() => _ = session.Query<Role>()
+				             .Where(r => !isActive.HasValue || r.IsActive == isActive.Value)
```

Comment:
Accessing `.Value` on a null parameter remains unsupported, but that is not a regression. And it is a minimal priority issue in my opinion, since this `.Value` access is not needed and has not any advantage as far as I know.